parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java

```diff
@@ -349,12 +351,16 @@ public void trackPlay(
                 videoEngagementManager = null;
             }
         }
+        @NonNull final String uuid = generatePixelId();
 
         // Enqueue the videostart
-        enqueueEvent(buildEvent(url, urlRef, "videostart", videoMetadata, extraData));
+        @NonNull final Map<String, Object> videostartEvent = buildEvent(url, urlRef, "videostart", videoMetadata, extraData);
+        videostartEvent.put(VIDEO_START_ID_KEY, uuid);
```

Comment:
Suggestion (💡): Instead of explicitly using an extra `.put(VIDEO_START_ID_KEY, uuid)` call after the event is constructed and returned, wherever that might be required, and thus letting the caller decide the event's state, which can be prone to misusage, how about moving that logic with the `buildEvent(...)` method itself and avoid any such misusage?

```java
@NonNull
    private Map<String, Object> buildEvent(
            String url,
            String urlRef,
            String action,
            ParselyMetadata metadata,
            Map<String, Object> extraData,
            @Nullable String uuid) {

        ...

        if (action.equals("videostart") || action.equals("vheartbeat")) {
            event.put(VIDEO_START_ID_KEY, uuid);
        }

        return event;
    }
```

FYI: I also think that doing this makes it more visible and explicit and only `videostart` and `vheartbeat` type of events can have an associated `vsid` attached to them.

PS: I understand that this is adding an extra parameter to the method, and that would have been better if a builder pattern would to be used, but I still think it is better than letting callers decide that after the fact, when the even is already created and returned by the `buildEvent(...)` method. 🤔

Reply:
Sure, I can do that 👍 

> I understand that this is adding an extra parameter to the method

That was my concern when deciding on approach here, but keeping all "bulding blocks" in `buildEvent` method is probably indeed better!

Reply:
Thanks for c05858254631ac13f36b9bfa5b0ec384054054a9 @wzieba , but I think you just missed removing this `hbEvent.put(VIDEO_START_ID_KEY, uuid);` bit [here](https://github.com/Parsely/parsely-android/blob/8de8a964124f03ca6c3d41c29224a660f30df1de/parsely/src/main/java/com/parsely/parselyandroid/ParselyTracker.java#L362), right? 🤔

Reply:
Yes, thank you! 🙏